### PR TITLE
Refine Op(new_ones, new_empty, new_empty_strided, new_full, new_zeros) | feat(torchlib)

### PR DESCRIPTION
**Pull request by xiaowuhu**
Refine ops

### CODE REVIEW

File: onnxscript/function_libs/torch_lib/ops/core.py [hunk 1]
```diff
@@ -4512,34 +4512,45 @@ def aten_new_full(
...
     """new_ones(Tensor self, SymInt[] size, *, ScalarType? dtype=None, Layout? layout=None, Device? device=None, bool? pin_memory=None) -> Tensor"""
 
-    one = op.Constant(value_float=1.0)
-    return op.Expand(one, size)
+    size1 = op.Size(op.Shape(self))
+    if size1 == 0:
+        one = op.Constant(value_float=1.0)
+    else:
+        size2 = op.Size(op.Shape(self))
+        one = op.Cast(size1 / size2, to=FLOAT.dtype)
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_lib/ops/core.py:4520]: As discussed, we can try using CastLike etc. to avoid the `if`

File: onnxscript/function_libs/torch_lib/ops/core.py [hunk 2]
```diff
@@ -4464,47 +4464,73 @@ def aten_negative(self: TensorType) -> TensorType:
     raise NotImplementedError()
 
 
-@torch_op("aten::new_empty", trace_only=True)
-def aten_new_empty(self: TTensor, size: INT64, dtype: int = -1) -> TTensor:
```
Comment (justinchuby) [inline onnxscript/function_libs/torch_lib/ops/core.py:4468]: Why is this change needed?

Comment (xiaowuhu) [reply onnxscript/function_libs/torch_lib/ops/core.py:4468]: add aten_new_empty_dtype overload

Comment (justinchuby) [reply onnxscript/function_libs/torch_lib/ops/core.py:4468]: I see. Thanks!

**Review by justinchuby — APPROVED** [review]
